pkg/controller/datadogagent/utils.go

```diff
@@ -544,7 +544,7 @@ func getEnvVarsForAgent(dda *datadoghqv1alpha1.DatadogAgent) ([]corev1.EnvVar, e
 				ValueFrom: getClusterAgentAuthToken(dda),
 			},
 		}
-		if *spec.ClusterAgent.Config.ClusterChecksEnabled {
+		if spec.ClusterAgent.Config.ClusterChecksEnabled != nil && *spec.ClusterAgent.Config.ClusterChecksEnabled {
```

Comment:
can we use `datadoghqv1alpha1.BoolValue()` here ?

Reply:
Yes we can !